packages/bsky/src/api/app/bsky/feed/getListFeed.ts

```diff
@@ -0,0 +1,131 @@
+import { Server } from '../../../../lexicon'
+import { QueryParams } from '../../../../lexicon/types/app/bsky/feed/getListFeed'
+import { FeedKeyset, getFeedDateThreshold } from '../util/feed'
+import { paginate } from '../../../../db/pagination'
+import AppContext from '../../../../context'
+import { setRepoRev } from '../../../util'
+import { Database } from '../../../../db'
+import {
+  FeedHydrationState,
+  FeedRow,
+  FeedService,
+} from '../../../../services/feed'
+import { ActorService } from '../../../../services/actor'
+import { GraphService } from '../../../../services/graph'
+import { createPipeline } from '../../../../pipeline'
+
+export default function (server: Server, ctx: AppContext) {
+  const getListFeed = createPipeline(
+    skeleton,
+    hydration,
+    noBlocksOrMutes,
+    presentation,
+  )
+  server.app.bsky.feed.getListFeed({
+    auth: ctx.authOptionalVerifier,
+    handler: async ({ params, auth, res }) => {
+      const viewer = auth.credentials.did
+      const db = ctx.db.getReplica()
+      const actorService = ctx.services.actor(db)
+      const feedService = ctx.services.feed(db)
+      const graphService = ctx.services.graph(db)
+
+      const [result, repoRev] = await Promise.all([
+        getListFeed(
+          { ...params, viewer },
+          { db, actorService, feedService, graphService },
+        ),
+        actorService.getRepoRev(viewer),
+      ])
+
+      setRepoRev(res, repoRev)
+
+      return {
+        encoding: 'application/json',
+        body: result,
+      }
+    },
+  })
+}
+
+export const skeleton = async (
+  params: Params,
+  ctx: Context,
+): Promise<SkeletonState> => {
+  const { list, cursor, limit } = params
+  const { db } = ctx
+  const { ref } = db.db.dynamic
+
+  const keyset = new FeedKeyset(ref('post.sortAt'), ref('post.cid'))
+  const sortFrom = keyset.unpack(cursor)?.primary
+
+  let builder = ctx.feedService
+    .selectPostQb()
+    .innerJoin('list_item', 'list_item.subjectDid', 'post.creator')
+    .where('list_item.listUri', '=', list)
+    .where('post.sortAt', '>', getFeedDateThreshold(sortFrom, 3))
+
+  builder = paginate(builder, {
+    limit,
+    cursor,
+    keyset,
+    tryIndex: true,
+  })
+  const feedItems = await builder.execute()
```

Comment:
This looks straightforward enough to to test-out w/ real data, may be worth a quick run!